fix: Remove `try_array_to_vec` and `downcast` (#82)
matrix-org/matrix-rust-sdk-crypto-wasm: CHANGELOG.md

# matrix-sdk-crypto-wasm unreleased

> -   Functions/methods that take a JavaScript `Array` as argument now delete the
>     array's items once their return, for example `Device::requestVerification`,

Reviewer: AIUI, it's not that the entries are deleted from the array, but rather that the individual items are themselves invalidated so that they can no longer be used?

Also: I think it happens as soon as the method is called, rather than once it returns? In other words, I can't do this:

```javascript
    const user = new UserId("@user:domain");
    await Promise.all([
         machine.query_keys_for_users([user]),
         // `user` is used below before `query_keys_for_users` returns, but AIUI this will fail.
         machine.update_tracked_users([user]),
     ]);
```

```suggestion
-   Functions/methods that take a JavaScript `Array` as argument now invalidate the
    items within that array so that they cannot be re-used. Affected methods include `Device::requestVerification`,
```

    `OlmMachine::updateTrackedUsers`, `OlmMachine::shareRoomKey`,

>     `OlmMachine::queryKeysForUsers`, `OlmMachine:getMissingSessions` and so on.

Reviewer: I think we need to give a complete list of the affected methods, so that applications can audit their usages.

Reviewer: actually, let's not bother with this. People can just go and look at the linked pull request.

**BREAKING CHANGES**

-   Rename `OlmMachine.init_from_store` introduced in v3.6.0 to
    `OlmMachine.initFromStore`.
    ([#84](https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/84))

> **BREAKING CHANGES**
> 

Reviewer: we now have two BREAKING CHANGES headings. (This is why I prefer tools like towncrier than maintaining changelog files in PRs, but I digress.)

```suggestion
```

-   Functions/methods that take a JavaScript `Array` as argument now invalidate the
    items within that array so that they cannot be re-used as soon as
    they are received by the functions/methods. See the patch for affected methods.
    ([#82](https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/82/))

> -   Update `wasm-bindgen` to 0.2.89. It allows to remove the `downcast` method.
>     It fixes [#51](https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/51),
>     thus the resulting JavaScript code of `matrix-rust-sdk-crypto-wasm` can
>     be minified with no issue now.
>     ([#82](https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/82/))

Reviewer: This is not a breaking change.

```suggestion
**Other changes**

-   Update `wasm-bindgen` to 0.2.89. It allows to remove the `downcast` method.
    It fixes [#51](https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/51),
    thus the resulting JavaScript code of `matrix-rust-sdk-crypto-wasm` can
    be minified with no issue now.
    ([#82](https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/82/))
```
